api/download.py

```diff
@@ -315,3 +315,92 @@ def download(self):
+            sessions = config.db.sessions.find({'project': req['_id']}, {'_id': 1})
+            session_ids = [s['_id'] for s in sessions]
+
+            # Grab acquisitions and their ids
+            acquisitions = config.db.acquisitions.find({'session': {'$in': session_ids}}, {'_id': 1})
+            acquisition_ids = [a['_id'] for a in acquisitions]
+            parent_ids = [req['_id']] + session_ids + acquisition_ids
+
+            # # Grab analyses and their ids
+            # analysis_ids = [an['_id'] for an in config.db.analyses.find({'parent.id': {'$in': parent_ids}})]
+
+            # for each type of container below it will have a slightly modified match query
+            cont_query = {
+                'projects': {'_id': req['_id']},
+                'sessions': {'project': req['_id']},
+                'acquisitions': {'session': {'$in': session_ids}},
+                'analyses': {'parent.id': {'$in': parent_ids}}
+            }
+        elif level == 'sessions':
+
+            # Grab acquisitions and their ids
+            acquisitions = config.db.acquisitions.find({'session': req['_id']}, {'_id': 1})
+            acquisition_ids = [a['_id'] for a in acquisitions]
+            parent_ids = [req['_id']] + acquisition_ids
+
+            # # Grab analyses and their ids
+            # analysis_ids = [an['_id'] for an in config.db.analyses.find({'parent.id': {'$in': parent_ids}})]
+
+            # for each type of container below it will have a slightly modified match query
+            cont_query = {
+                'sessions': {'_id': req['_id']},
+                'acquisitions': {'session': req['_id']},
+                'analyses': {'parent.id': {'$in': parent_ids}}
+            }
+            containers = containers[1:]
+        elif level == 'acquisitions':
+
+            cont_query['acquisitions'] = {'_id': req['_id']}
+            containers = ['acquisitions']
+        elif level == 'analyses':
+            cont_query['analyses'] = {'_id': req['_id']}
+            containers = containers[-1:]
+        else:
+            self.abort(400, "{} not a recognized level".format(level)) 
+
+        res = {}
+        for cont_name in containers:
+            # Aggregate file types
+            pipeline = [
+                {'$match': cont_query[cont_name]},
+                {'$unwind': '$files'},
+                {'$project': {'_id': '$_id', 'type': '$files.type','mbs': {'$divide': ['$files.size', BYTES_IN_MEGABYTE]}}},
+                {'$group': {
+                    '_id': '$type',
+                    'count': {'$sum' : 1}, 
+                    'mb_total': {'$sum':'$mbs'}, 
+                    'nodes' : {
+                        '$addToSet': {'level': {'$literal':cont_name}, '_id': '$_id'}
+                    }
+                }}
+            ]
+
+            try:
+                result = config.db.command('aggregate', cont_name, pipeline=pipeline)
+            except Exception as e: # pylint: disable=broad-except
```

Comment:
The string version of the failure would get returned to the API user. In situations like this it's better to internally log the error (`log.warning(e)`) and then `self.abort(500, 'Failure to load summary')` or something like that so the user doesn't get sent a mongo error.